From 39b2ee3b254753df2848d4c405d986fbd530dbff Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Mon, 5 Oct 2015 16:03:33 +0530
Subject: [PATCH] USBC/Src_pe: Sink caps received can be processed in any
 state.

Some devices provides sink caps without source port request.
Hence source pe should be able to process the sink caps in any state.

Change-Id: Iae8f76027031693f98fea296f97c33e653435d17
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-19347
---
 drivers/usb/typec/pd/src_port_pe.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/typec/pd/src_port_pe.c b/drivers/usb/typec/pd/src_port_pe.c
index 1e02820..838791f 100644
--- a/drivers/usb/typec/pd/src_port_pe.c
+++ b/drivers/usb/typec/pd/src_port_pe.c
@@ -617,11 +617,11 @@ src_pe_rcv_pkt(struct policy *srcp, struct pd_packet *pkt, enum pe_event evt)
 		if (src_pe->state == PE_SRC_GET_SINK_CAP) {
 			mutex_lock(&src_pe->pe_lock);
 			src_pe->state = SRC_PE_STATE_PD_CONFIGURED;
-			src_pe->got_snk_caps = 1;
 			mutex_unlock(&src_pe->pe_lock);
 			complete(&src_pe->srt_complete);
-			src_pe_handle_snk_cap_rcv(src_pe, pkt);
 		}
+		src_pe->got_snk_caps = 1;
+		src_pe_handle_snk_cap_rcv(src_pe, pkt);
 		break;
 	case PE_EVT_RCVD_DR_SWAP:
 		src_pe_handle_rcv_dr_swap(src_pe);
-- 
1.9.1

